pydrive2/files.py

```diff
@@ -345,19 +350,35 @@ def download(fd, request):
             except errors.HttpError as error:
                 exc = ApiRequestError(error)
                 if (
-                    exc.error["code"] != 403
-                    or exc.GetField("reason") != "fileNotDownloadable"
+                    exc.error["code"] == 403
+                    and exc.GetField("reason") == "fileNotDownloadable"
+                ):
+                    mimetype = mimetype or "text/plain"
+                    fd.seek(0)  # just in case `download()` modified `fd`
+                    try:
+                        download(
+                            fd,
+                            files.export_media(
+                                fileId=file_id, mimeType=mimetype
+                            ),
+                        )
+                    except errors.HttpError as error:
+                        raise ApiRequestError(error)
+                elif (
+                    exc.error["code"] == 416
+                    and exc.GetField("reason")
+                    == "requestedRangeNotSatisfiable"
                 ):
+                    try:
+                        # check if it is an empty file
+                        request = files.get_media(fileId=file_id)
```

Comment:
This is adding a yet another API call, so probably not worth bothering with. We can probably just ignore this particular code/reason.